tests/cpp/dataLevel0BlocksMemory_test.cpp

```diff
@@ -0,0 +1,175 @@
+#include "../../hnswlib/hnswlib.h"
+
+#include <assert.h>
+
+#include <vector>
+#include <iostream>
+#include <cstdio>
+#include <thread>
+#include <chrono>
+
+namespace {
+
+const size_t M = 32;
+const size_t ef_construction = 500;
+const size_t random_seed = 100;
+const bool allow_replace_deleted = false;
+
+const size_t dimension = 1024;
+const size_t total_items = 100 * 10000;
+const size_t num_query = 500 * 10000;
+size_t topk = 10;
+const size_t max_thread_num = 48;
+const std::string index_path = "./hnsw.index";
+
+std::vector<float> data(total_items * dimension);
+std::vector<float> query(num_query * dimension);
+
+
+void check_knn_closer(hnswlib::AlgorithmInterface<float>* alg_hnsw) {
+    for (size_t j = 0; j < num_query; ++j) {
+        const void* p = query.data() + j * dimension;
+        auto gd = alg_hnsw->searchKnn(p, topk);
+        auto res = alg_hnsw->searchKnnCloserFirst(p, topk);
+        assert(gd.size() == res.size());
+        size_t t = gd.size();
+        while (!gd.empty()) {
+            assert(gd.top() == res[--t]);
+            gd.pop();
+        }
+    }
+    std::cout << "test hnsw search knn closer first success..." << std::endl;
+}
+
+void test_compatibility(bool hnsw_first_use_blocks_memory,
+                        bool hnsw_second_use_blocks_memory) {
+
+    std::cout << "================== test compatibility ==================" << std::endl;
+    hnswlib::L2Space space(dimension);
+    hnswlib::AlgorithmInterface<float>* alg_hnsw_first = new hnswlib::HierarchicalNSW<float>(&space, 2 * total_items,
+            M, ef_construction, random_seed, allow_replace_deleted, hnsw_first_use_blocks_memory);
+
+    for (size_t i = 0; i < total_items; ++i) {
+        alg_hnsw_first->addPoint(data.data() + dimension * i, i);
+    }
+    check_knn_closer(alg_hnsw_first);
```

Comment:
I'm not sure what we are supposed to check here. `searchKnnCloserFirst` only calls `searchKnn` and rearranges the found elements."